src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -46,6 +46,22 @@ public class ObservableCollection<T> : Collection<T>, INotifyCollectionChanged,
         /// <exception cref="ArgumentNullException"> collection is a null reference </exception>
         public ObservableCollection(IEnumerable<T> collection) : base(CreateCopy(collection)) { }
 
+        /// <summary>
+        /// Initializes a new instance of the ObservableCollection class
+        /// that contains elements copied from the specified list
+        /// </summary>
+        /// <param name="list">The list whose elements are copied to the new list.</param>
+        /// <remarks>
+        /// The elements are copied onto the ObservableCollection in the
+        /// same order they are read by the enumerator of the list.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException"> list is a null reference </exception>
+        public ObservableCollection(List<T> list)
+            : base((list != null) ? new List<T>(list.Count) : list)
+        {
+            CreateCopy(list);
+        }
```

Comment:
I will make the changes you point out above.

> Also, are there tests for both constructors?

Tere is [a test](https://github.com/dotnet/corefx/blob/master/src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_ConstructorAndPropertyTests.cs#L31-L46) that already will handle testing both constructors which is why I didn't add any more.